pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -2492,52 +2505,35 @@ func fetchRegistrationEntry(ctx context.Context, db *sqlDB, entryID string) (*co
 	}
 	defer rows.Close()
 
-	var entry *common.RegistrationEntry
-	for rows.Next() {
-		var r entryRow
-		if err := scanEntryRow(rows, &r); err != nil {
-			return nil, err
-		}
-
-		if entry == nil {
-			entry = new(common.RegistrationEntry)
-		}
-		if err := fillEntryFromRow(entry, &r); err != nil {
-			return nil, err
-		}
-	}
-
-	if err := rows.Err(); err != nil {
-		return nil, newWrappedSQLError(err)
-	}
-
-	return entry, nil
+	var entries []*common.RegistrationEntry
+	entries, _, err = rowsToCommonRegistrationEntries(rows, entries)
+	return entries, err
 }
 
-func buildFetchRegistrationEntryQuery(dbType string, supportsCTE bool, entryID string) (string, []any, error) {
+func buildFetchRegistrationEntryQuery(dbType string, supportsCTE bool, entryIDs []string) (string, []any, error) {
```

Comment:
```suggestion
func buildFetchRegistrationEntriesQuery(dbType string, supportsCTE bool, entryIDs []string) (string, []any, error) {
```

similarly for the dialect specific ones.